ocs_ci/deployment/hosted_cluster.py

```diff
@@ -260,12 +260,50 @@ def deploy_hosted_ocp_clusters(self, cluster_names_list=None):
             # operators and finish the rest preparation steps. For the rest of the clusters we will only deploy OCP
             # with hcp command.
             first_ocp_deployment = index == 0
-            cluster_name = hosted_ocp_cluster.deploy_ocp(
-                deploy_cnv=first_ocp_deployment,
-                deploy_acm_hub=first_ocp_deployment,
+
+            # Put logic on checking and deploying dependencies here
+            if first_ocp_deployment:
+                # ACM installation is a default for Provider/Converged deployments
+                deploy_acm_hub = config.ENV_DATA.get("deploy_acm_hub_cluster", True)
+                # CNV installation is a default for Provider/Converged deployments
+                deploy_cnv = config.DEPLOYMENT.get("cnv_deployment", True)
+                deploy_mce = config.DEPLOYMENT.get("mce_deployment", False)
+                deploy_hyperconverged = config.ENV_DATA.get(
+                    "deploy_hyperconverged", False
+                )
+
+                # Validate conflicting deployments
+                if deploy_acm_hub and deploy_mce:
+                    raise UnexpectedDeploymentConfiguration(
+                        "Conflict: Both 'deploy_acm_hub_cluster' and 'mce_deployment' are enabled. Choose one."
+                    )
+                if deploy_cnv and deploy_hyperconverged:
+                    raise UnexpectedDeploymentConfiguration(
+                        "Conflict: Both 'cnv_deployment' and 'deploy_hyperconverged' are enabled. Choose one."
+                    )
+
+            else:
+                deploy_acm_hub = False
```

Comment:
Shouldn't the values be collected from config even after first deployment?

Reply:
this function is driven by `ocs_ci/deployment/deployment.py:deploy_cluster` which has its own `deploy_cnv` call.
We have it here in case we don't want to deploy Provider and use fixture, or try to deploy clients directly from other place.

So in this case it passes the calls where we do install cnv - mandatory, when config saying this and now only check that with first client cnv and rest dependencies are installed.
Hope that answers on q.